repository_service_tuf/cli/artifact/repository.py

```diff
@@ -0,0 +1,278 @@
+                    f"Repository {repository} has incorrect configuration. "
+                    "Please verify you're using proper types:\n"
+                    "artifact_base_url: <string>\n"
+                    "hash_prefix: <bool>\n"
+                    "metadata_url: <string>\n"
+                    "trusted_root: <base64 string>\n"
+                )
+        else:
+            raise click.ClickException(
+                f"Repository {repository} is missing in your configuration"
+            )
+    else:
+        if rstuf_config.get("REPOSITORIES"):
+            is_default = ""
+            for repo in rstuf_config.get("REPOSITORIES").keys():
+                if current_repository and current_repository == repo:
+                    style = "b green"
+                    is_default = " (default)"
+                else:
+                    style = "b white"
+                    is_default = ""
+                console.print(f"{repo}{is_default}", style=style)
+        else:
+            raise click.ClickException("There are no configured repositories")
+
+
+@repository.command()
+@click.pass_context
+@click.argument("repository")
+def set(context: Context, repository: str) -> None:
+    """
+    Switch current repository.
+    """
+
+    rstuf_config = context.obj.get("settings").as_dict()
+
+    rstuf_config["CURRENT_REPOSITORY"] = repository
+    if context.obj.get("config"):
+        write_config(context.obj.get("config"), rstuf_config, False)
+        console.print(f"Current repository changed to {repository}")
+
+
+@repository.command()
+@click.option(
+    "-n",
+    "--name",
+    help="A repository name.",
+    type=str,
+    required=True,
+    default=None,
+)
+@click.option(
+    "-r",
+    "--root",
+    help="A metadata URL to the initial trusted root or a local file.",
+    type=str,
+    required=True,
+    default=None,
+)
+@click.option(
+    "-m",
+    "--metadata-url",
+    help="TUF Metadata repository URL.",
+    type=str,
+    required=True,
+    default=None,
+)
+@click.option(
+    "-a",
+    "--artifacts-url",
+    help="An artifacts base URL to fetch from.",
+    type=str,
+    required=True,
+    default=None,
+)
+@click.option(
+    "-p",
+    "--hash-prefix",
+    help="A hash prefix.",
+    is_flag=True,
+    required=None,
+)
+@click.pass_context
+def add(
+    context: Context,
+    name: str,
+    root: str,
+    metadata_url: str,
+    artifacts_url: str,
+    hash_prefix: bool,
+) -> None:
+    """
+    Add a new repository.
+    """
+
+    rstuf_config = context.obj.get("settings").as_dict()
+
+    success_msg: str = f"Successfully added {name} repository to config"
+    if rstuf_config.get("REPOSITORIES") and rstuf_config["REPOSITORIES"].get(
+        name
+    ):
+        console.print(f"Repository {name} already configured")
+        success_msg = f"Successfully updated repository {name}"
+
+    encoded_root = b""
+    if root:
+        encoded_root = base64.b64encode(bytes(root, "utf-8"))
```

Comment:
Why did you decide to encode the root?

Reply:
We wouldn't like to keep the trusted root path in plane text in our config